fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -614,4 +617,37 @@ private static URI checksumURI( final String checksum ) {
         }
         return null;
     }
+
+    protected void createFixityEvent(final EventBus eventBus, final RdfStream rs,
+                                     final String baseURL, final String agent,
+                                     final String userID) {
+        final FixityEvent fixityEvent = new FixityEvent();
+        fixityEvent.setDate(System.currentTimeMillis());
+        fixityEvent.setType(128);
+        final StmtIterator it = rs.asModel().listStatements();
+        final HashMap<String,String> info = new HashMap<String,String>();
+        while (it.hasNext()) {
+           final Statement st = it.next();
+           if (st.getPredicate().toString().equals("http://www.loc.gov/premis/rdf/v1#hasFixity")) {
+               String identifier = st.getObject().toString().replaceAll(baseURL,"");
+               identifier = "/" + identifier.substring(0,identifier.indexOf("#fixity"));
+               fixityEvent.setPath(identifier);
+               fixityEvent.setIdentifier(identifier);
+           }
+            if (st.getPredicate().toString().equals("http://www.loc.gov/premis/rdf/v1#hasSize")) {
+                info.put(st.getPredicate().toString(), st.getObject().toString());
+            }
+            if (st.getPredicate().toString().equals("http://www.loc.gov/premis/rdf/v1#hasMessageDigest")) {
+                info.put(st.getPredicate().toString(),st.getObject().toString());
+            }
+            if (st.getPredicate().toString().equals("http://fedora.info/definitions/v4/repository#status")) {
+                info.put(st.getPredicate().toString(),st.getObject().toString());
+            }
```

Comment:
This whole section is not good. It ignores the fact that an `RdfStream` _is_ an iterator, expensively converts it to a `Model`, then iterates over the `Statement`s in the `Model`! Please just iterate over the `RdfStream` to begin with.